_scripts/getRegions.mjs

```diff
@@ -0,0 +1,128 @@
+
+import { mkdirSync, readFileSync, rmSync, writeFileSync } from 'fs'
+import { dirname } from 'path'
+import { fileURLToPath } from 'url'
+import { Innertube, Misc } from 'youtubei.js'
+
+const STATIC_DIRECTORY = `${dirname(fileURLToPath(import.meta.url))}/../static`
+
+const activeLanguagesPath = `${STATIC_DIRECTORY}/locales/activeLocales.json`
+/** @type {string[]} */
+const activeLanguages = JSON.parse(readFileSync(activeLanguagesPath, { encoding: 'utf8' }))
+
+// en-US is en on YouTube
+const initialResponse = await scrapeLanguage('en')
+
+// Scrape language menu in en-US
+
+/** @type {string[]} */
+const youTubeLanguages = initialResponse.data.actions[0].openPopupAction.popup.multiPageMenuRenderer.sections[1].multiPageMenuSectionRenderer.items[1].compactLinkRenderer.serviceEndpoint.signalServiceEndpoint.actions[0].getMultiPageMenuAction.menu.multiPageMenuRenderer.sections[0].multiPageMenuSectionRenderer.items
+  .map(({ compactLinkRenderer }) => {
+    return compactLinkRenderer.serviceEndpoint.signalServiceEndpoint.actions[0].selectLanguageCommand.hl
+  })
+
+// map FreeTube languages to their YouTube equivalents
+
+const foundLanguageNames = ['en-US']
+const unusedYouTubeLanguageNames = []
+const languagesToScrape = []
+
+for (const language of youTubeLanguages) {
+  if (activeLanguages.includes(language)) {
+    foundLanguageNames.push(language)
+    languagesToScrape.push({
+      youTube: language,
+      freeTube: language
+    })
+  } else if (activeLanguages.includes(language.replace('-', '_'))) {
+    const withUnderScore = language.replace('-', '_')
+    foundLanguageNames.push(withUnderScore)
+    languagesToScrape.push({
+      youTube: language,
+      freeTube: withUnderScore
+    })
+  }
+  // special cases
+  else if (language === 'de') {
+    foundLanguageNames.push('de-DE')
+    languagesToScrape.push({
+      youTube: 'de',
+      freeTube: 'de-DE'
+    })
+  } else if (language === 'fr') {
+    foundLanguageNames.push('fr-FR')
+    languagesToScrape.push({
+      youTube: 'fr',
+      freeTube: 'fr-FR'
+    })
+  } else if (language === 'no') {
+    // according to https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes
+    // "no" is the macro language for "nb" and "nn"
+    foundLanguageNames.push('nb_NO', 'nn')
+    languagesToScrape.push({
+      youTube: 'no',
+      freeTube: 'nb_NO'
+    })
+    languagesToScrape.push({
+      youTube: 'no',
+      freeTube: 'nn'
+    })
+  } else if (language !== 'en') {
+    unusedYouTubeLanguageNames.push(language)
+  }
+}
+
+console.log("Active FreeTube languages that aren't available on YouTube:")
+console.log(activeLanguages.filter(lang => !foundLanguageNames.includes(lang)).sort())
+
+console.log("YouTube languages that don't have an equivalent active FreeTube language:")
+console.log(unusedYouTubeLanguageNames.sort())
+
+// Scrape the location menu in various languages and write files to the file system
+
+rmSync(`${STATIC_DIRECTORY}/geolocations`, { recursive: true })
```

Comment:
I feel like we shouldn't remove the directory and should just overwrite matching files. This will allow us to add other files that are supported by FreeTube but not by YouTube (ex: notice how some files were removed without a new version)

Reply:
Some of the ones that were removed were for languages that we don't even have in FreeTube, not even as inactive ones like `ee` or `zh` (we have `zh-CN` and `zh-TW`, which both have YouTube equivalents).

The main issue with maintaining our own files, is that when they inevitably become outdated (YouTube adding and removing regions), you will be able to select non-existent regions (brings us back to the issue this pull request is meant to solve) and won't be able to select new ones until they are translated.